### PR TITLE
ci: Implement CI/CD for SDK 

### PR DESCRIPTION
**Pull request by kouvaliasnick**
Introducing CI/CD using semantic-releases. Trigger pre-releases on PR merges in develop based on the PR title which has to be written using the semantic convention. Every pre-release will trigger in its turn the update of the SDK version in creek using dispatch events so integration tests will run to always be sure that Creek works well with the SDK. When is time for release, a PR needs to be created from develop to the main branch. Once the PR is merged the actual release will be triggered.

### CODE REVIEW

File: .github/workflows/pre-release.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Install dependencies
+        run: |
+          npm install -g npm@8.5.2 node-pre-gyp
```
Comment (AntonNeld) [inline .github/workflows/pre-release.yml:28]: Why is this needed?

File: .github/workflows/notify-sdk-update.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: Notify SDK update
+on:
+  release:
+    types: [published]
```
[the remark this answers is not in the document]
Comment (kouvaliasnick) [reply .github/workflows/notify-sdk-update.yml:4]: I'm counting on that, but only way to actual test this is to merge this PR

File: package.json [hunk 3]
```diff
@@ -16,14 +16,32 @@
     "style-write": "appland-style --browser --write src/",
     "style-check": "appland-style --browser --check src/",
     "postversion": "scripts/postversion_hook.sh",
-    "prepare": "husky install || exit 0"
+    "prepare": "husky install || exit 0",
+    "semantic-release": "semantic-release"
```
[the remark this answers is not in the document]
Comment (kouvaliasnick) [reply package.json:20]: after the discussion we had seems that the advantages (not having to thing about versioning, self-explanatory commit history, utilizing the GH releases, automated change-logs, notifications on PRs that are included in the release etc.) outweigh the disadvantage of having the friction with properly renaming the PR title before squash and merge.

File: .github/workflows/pre-release.yml (hunk 1 above)
Comment (kouvaliasnick) [reply .github/workflows/pre-release.yml]: thanks for pointing that out. It is not needed, It was a left over of trying to implement npm cache but we can do that in another PR

**Review by AntonNeld — APPROVED** [review]
